dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2426,10 +2426,16 @@ public Node visitEnhancedForLoop(EnhancedForLoopTree tree, Void p) {
             MethodInvocationNode nextCallNode =
                     new MethodInvocationNode(
                             nextCall, nextAccessNode, Collections.emptyList(), getCurrentPath());
+            // If the type of iteratorVariable is a capture, its type tree may be missing
+            // annotations, so save the expression in the node so that the full type can be found
+            // later.
+            nextCallNode.setIterExpression(expression);
             nextCallNode.setInSource(false);
             extendWithNode(nextCallNode);
 
-            translateAssignment(variable, new LocalVariableNode(variable), nextCall);
+            AssignmentNode assignNode =
+                    translateAssignment(variable, new LocalVariableNode(variable), nextCall);
+            ((MethodInvocationNode) assignNode.getExpression()).setIterExpression(expression);
```

Comment:
Can you add a comment why this second call is necessary? I would have expected the expression to be equal to `nextCallNode`, on which `setIterExpression` was already called.